### PR TITLE
Fixed #14098 -- Prevented crash for introspection errors in inspectdb

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: django/core/management/commands/inspectdb.py [hunk 1]
```diff
@@ -57,25 +58,32 @@ def strip_prefix(s):
...
+                        indexes = connection.introspection.get_indexes(cursor, table_name)
+                    except NotImplementedError:
+                        indexes = {}
+                    try:
+                        constraints = connection.introspection.get_constraints(cursor, table_name)
+                    except NotImplementedError:
+                        constraints = {}
+                    table_description = connection.introspection.get_table_description(cursor, table_name)
+                except Exception as e:
+                    yield "# Unable to inspect table %s" % table_name
```
Comment (timgraham) [inline django/core/management/commands/inspectdb.py]: %s -> %r so we get quotes?

File: tests/inspectdb/tests.py [hunk 2]
```diff
@@ -268,3 +269,17 @@ def test_custom_fields(self):
             self.assertIn("big_int_field = models.BigIntegerField()", output)
         finally:
             connection.introspection.data_types_reverse = orig_data_types_reverse
+
+    def test_introspection_errors(self):
+        """
+        Introspection errors should not crash the command, but the error should
```
Comment (timgraham) [inline tests/inspectdb/tests.py]: but -> and

File: tests/inspectdb/tests.py [hunk 3]
```diff
@@ -268,3 +269,17 @@ def test_custom_fields(self):
...
+        be visible in the output.
+        """
+        out = StringIO()
+        with mock.patch('django.db.backends.base.introspection.BaseDatabaseIntrospection.table_names',
+                        return_value=['notexist']):
```
Comment (timgraham) [inline tests/inspectdb/tests.py]: "nonexistent" would be the word

File: django/core/management/commands/inspectdb.py (hunk 1 above)
Comment (claudep) [inline django/core/management/commands/inspectdb.py]: Or directly add quotes in the string, as %r might produce a u'' prefix on Python 2.

Comment (timgraham) [inline django/core/management/commands/inspectdb.py]: Yes, that's better.

**Comment by timgraham** [comment]
I don't feel too strongly about whether or not to document it. It might be useful to know in advance that errors might show up in the output I suppose.

**Comment by claudep** [comment]
Not convinced either. This is how things should work in the first place, isn't it? (errors are reported)

**Pull request merged**